src/include/common/task_system/task.h

```diff
@@ -72,6 +72,12 @@ class Task {
         return exceptionsPtr;
     }
 
+    // Resets a task so that it can be re-used and run again
+    virtual void reset() {
```

Comment:
Normally I would avoid using `reset` as a function name because there is also `smart_pointer->reset()` which might cause confusion. Consider using `resetState` or some other naming.